library-model/library-model-generator-integration-test/src/test/java/com/uber/nullaway/libmodel/LibraryModelIntegrationTest.java

```diff
@@ -0,0 +1,76 @@
+package com.uber.nullaway.libmodel;
+
+import com.google.errorprone.CompilationTestHelper;
+import com.uber.nullaway.NullAway;
+import java.util.Arrays;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class LibraryModelIntegrationTest {
+
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private CompilationTestHelper compilationHelper;
+
+  @Before
+  public void setup() {
+    compilationHelper = CompilationTestHelper.newInstance(NullAway.class, getClass());
+  }
+
+  @Test
+  public void libraryModelNullableReturnsTest() {
+    compilationHelper
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:JarInferEnabled=true",
+                "-XepOpt:NullAway:JarInferUseReturnAnnotations=true"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import com.uber.nullaway.libmodel.AnnotationExample;",
+            "class Test {",
+            "  static AnnotationExample annotationExample = new AnnotationExample();",
+            "  static void test(String value){",
+            "  }",
+            "  static void testPositive() {",
+            "    // BUG: Diagnostic contains: passing @Nullable parameter 'annotationExample.makeUpperCase(\"nullaway\")'",
+            "    test(annotationExample.makeUpperCase(\"nullaway\"));",
+            "  }",
+            "  static void testNegative() {",
+            "    test(annotationExample.nullReturn());",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void libraryModelNullableReturnsArrayTest() {
+    compilationHelper
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:JarInferEnabled=true",
+                "-XepOpt:NullAway:JarInferUseReturnAnnotations=true"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import com.uber.nullaway.libmodel.AnnotationExample;",
+            "class Test {",
+            "  static AnnotationExample annotationExample = new AnnotationExample();",
+            "  static void test(Integer[] value){",
+            "  }",
+            "  static void testPositive() {",
+            "    // BUG: Diagnostic contains: passing @Nullable parameter 'annotationExample.generateIntArray(7)'",
+            "    test(annotationExample.generateIntArray(7));",
+            "  }",
+            "}")
+        .doTest();
+  }
```

Comment:
Add a test case without the `-XepOpt:NullAway:JarInferEnabled` flag or whatever we end up calling it, to check that this functionality is off without it.

Reply:
Also, may I suggest a case with a model adding `@NonNull` to a method argument, just for completeness here?

Reply:
@lazaroclapp We have not added the functionality for processing the annotations on method parameters in this PR so a test case with `@NonNull` on a method parameter will not be supported on these changes.

Reply:
Also @lazaroclapp I don't think supporting `@NonNull` on parameters will be a priority.  Instead we are focusing on supporting treatment of a modeled class as `@NullMarked`, so everything is `@NonNull` by default, and then pulling in any explicit `@Nullable` annotations on parameters into the model

Reply:
There isn't a chance that specific classes will be `@NullUnmarked` with a few `@NonNull` locations? Either way, understood, not needed as part of this PR. I do feel in the medium term we might get surprised by the lack of symmetry here when using this for other kinds of library models.